deepchecks/nlp/checks/model_evaluation/single_dataset_performance.py

```diff
@@ -0,0 +1,144 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module containing the single dataset performance check."""
+from numbers import Number
+from typing import TYPE_CHECKING, Callable, Dict, List, TypeVar, Union
+
+import pandas as pd
+
+from deepchecks.core import CheckResult, ConditionCategory, ConditionResult
+from deepchecks.core.checks import ReduceMixin
+from deepchecks.core.errors import DeepchecksValueError
+from deepchecks.nlp.base_checks import SingleDatasetCheck
+from deepchecks.nlp.context import Context
+from deepchecks.nlp.metric_utils.scorers import infer_on_text_data
+from deepchecks.utils.docref import doclink
+from deepchecks.utils.strings import format_number
+
+if TYPE_CHECKING:
+    from deepchecks.core.checks import CheckConfig
+
+__all__ = ['SingleDatasetPerformance']
+
+
+SDP = TypeVar('SDP', bound='SingleDatasetPerformance')
+
+
+class SingleDatasetPerformance(SingleDatasetCheck, ReduceMixin):
+    """Summarize given model performance on a dataset based on selected scorers.
+
+    Parameters
+    ----------
+    scorers : Union[List[str], Dict[str, Union[str, Callable]]], default: None
+        List of scorers to use. If None, use default scorers.
+        Scorers can be supplied as a list of scorer names or as a dictionary of names and functions.
+    """
+
+    def __init__(self,
+                 scorers: Union[List[str], Dict[str, Union[str, Callable]]] = None,
+                 **kwargs):
+        super().__init__(**kwargs)
+        self.scorers = scorers
+
+    def run_logic(self, context: Context, dataset_kind) -> CheckResult:
+        """Run check."""
+        dataset = context.get_data_by_kind(dataset_kind)
+        model = context.model
+        scorers = context.get_scorers(self.scorers, use_avg_defaults=False)
+
+        results = []
+        classes = dataset.classes
+        for scorer in scorers:
+            scorer_value = infer_on_text_data(scorer, model, dataset)
+            if isinstance(scorer_value, Number):
+                results.append([pd.NA, scorer.name, scorer_value])
+            else:
+                results.extend(
+                    [[class_name, scorer.name, class_score]
+                     for class_score, class_name in zip(scorer_value, classes)])
```

Comment:
make sure classes are in same order as returned from sklearn metrics

Reply:
Classes are sorted (if not defined by user), and that is the way sklearn returns them